gradio/components/image_editor.py

```diff
@@ -117,6 +117,25 @@ def __post_init__(self):
             )
 
 
+@document()
+@dataclasses.dataclass
+class LayerOptions:
+    """
+    A dataclass for specifying options for the layer tool in the ImageEditor component. An instance of this class can be passed to the `layers` parameter of `gr.ImageEditor`.
+    Parameters:
+        allow_additional_layers: If True, users can add additional layers to the image. If False, the add layer button will not be shown.
+        layers: A list of layers to make available to the user when using the layer tool. One layer must be provided, if the length of the list is 0 then a layer will be generated automatically.
+    """
+
+    allow_additional_layers: bool = True
+    layers: list[str] | None = None
+    disabled: bool = False
```

Comment:
What does `disabled` do?

Reply:
Perhaps we should add a parameter to allow reordering of layers

Reply:
Disabled hides the layer menu (same as false), it's really just there because we convert Boolean's to an instance of `LayerOptions`. 

Regarding reordering, I did consider this but I'm not how useful that would be. Easy addition though, so happy to add it.

Reply:
Ah okay I was wondering because it was missing a docstring, but if it's only an internal parameter, perhaps we can rename to `_disabled`